### ISSUE TITLE
Conbee II + ZHA, IKEA bulbs reported as off despite being on

### ISSUE THREAD

**Issue by comatose-tortoise** [zigpy/zigpy-deconz#78]
**System is:**
Mini-ITX-board with Intel CPU / Ubuntu 16.04.06 / Docker / Home Assistant (0.100.2)
Conbee II firmware: deCONZ_ConBeeII_0x264a0700.bin.GCF
Extension USB cable to the Conbee II
Zigbee devices are a number of different IKEA Tradfri lights and two IKEA Tradfri power plugs, and one Xiaomi vibration sensor

**Problem:** 
The IKEA bulbs (of several different types, GU10, E27, E14, etc) get the wrong status indication in the HA GUI. If I turn the bulbs on, a number of them will relatively quickly become reported as off in the GUI, despite being on. If I "turn them on" in the GUI, nothing happens and they remain on, and if I then switch them off in the GUI, they turn off.

Interestingly, if I use my "group switches" (i.e. all bulbs in the kitchen), and turn them off, they do actually turn off, including any lights that are indicated as being off in the GUI. 

Anyway, this causes problems with automations, among other things. It's also a problem not being able to trust the status report in HA, but instead have to actually check the light IRL to know if it's off. 

This problem has been present through several different HA versions, from when I changed from the IKEA gateway to the Conbee II. The IKEA switches and the Xiaomi vibration sensor are not affected.

**Comment by Adminiuga** [comment]
do you get attribute reports on cluster 6 from those lights? Keep in mind you won't be getting attribute reports while the light is physically off and when power being restored the light might not send the report immediately, but should send one within the 15 min. Make sure you do get attribute reports while the light is being powered on at least once every 15 min.

**Comment by comatose-tortoise** [comment]
I don't know. How can I check those attribute reports on cluster 6?

**Comment by Adminiuga** [comment]
Light must have mains power. grep the log file for `'0x0006. Attribute report received:'` on_off=Bool.false`
You should see something like
```
 [0x32d4:1:0x0006] Attribute report received: on_off=Bool.false
```
where `0x32d4` is the NWK address of the device, `1` is the endpoint (may vary from vendor to vendor) and `0x0006` is ZCL Cluster 6 -- On/Off cluster.

**Comment by comatose-tortoise** [comment]
I have no "Attribute report received" at all in the log!

**Comment by Adminiuga** [comment]
Pick a few lights from config panel -> zha, click reconfigure device and post log here. Wait a few hours and check log again if you get attribute updates from the lights you reconfigured

**Comment by comatose-tortoise** [comment]
Apologies for the slow response.

Incidentally, a short while after I last posted here, I removed an IKEA plug that was constantly becoming "unavailable" a while after being added. As son as I removed the constantly failing plug from HA, the number of lights that got incorrect status indications in HA was reduced significantly. I don't know how these two things are connected, but it was obvious that they were.

That being said, the phenomena is not completely gone. I still get incorrect status indications in HA once in a while, but it's pretty rare now. I have been trying to see if certain bulbs are more affected than others, but it seems most of them report incorrectly once in a while. I will try to reconfigure one of them and look for attribute updates.

**Comment by comatose-tortoise** [comment]
Ok, now the lights are constantly reporting the wrong state again, messing up automations. I got the impression it started after I added a new devide to ZHA, the Philips Hue motion sensor. It works perfectly fine itself, but could there be a connection to the other problems I'm seeing?

I need some help to get the logs. Which logging do I need to enable to see if they report attributes?
Enabling
`    homeassistant.components.zha: debug
    zigpy: debug
    zigpy_deconz: debug`
Seems to show it, but generates ~1000 lines per minute.

**Comment by Adminiuga** [comment]
`0x0006. Attribute report received:` only if you do `grep '0x0006. Attribute report received:' home-assistant.log` if you do CTRL-F in a editor the line would look like `[0x32d4:1:0x0006] Attribute report received: on_off=Bool.false`

**Comment by comatose-tortoise** [comment]
I was using an editor, doing CTRL+F, but searching for `0x0006`, and the only results are these:
`2019-11-21 19:07:37 DEBUG (MainThread) [zigpy.zcl] [0x299f:1:0x0006] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=False> manufacturer=None tsn=86 command_id=Command.Default_Response>
2019-11-21 19:07:37 DEBUG (MainThread) [zigpy_deconz.api] 'aps_data_indication' response from <DeconzAddress address_mode=ADDRESS_MODE.NWK address=0x299f>, ep: 1, profile: 0x0104, cluster_id: 0x0006, data: b'08560b0100'
2019-11-21 19:07:37 DEBUG (MainThread) [homeassistant.components.zha.core.channels] [0x299f:1:0x0006]: executed command: on with args: () with kwargs: {} and result: [1, <Status.SUCCESS: 0>]`

**Comment by Adminiuga** [comment]
`[0x299f:1:0x0006]` -- 0x0006 is Cluster-ID 6 == On/Off cluster, `command_id=Command.Default_Response>` indicates this message is a response to a "cluster command" sent to the device. In other words this is a confirmation to a request turn off/on the light.

If you don't have any "Attribute Updated"  for cluster 0x0006 that means either the lights are not sending those (HUE lights) or you are not getting those for whatever reason.
If those are not HUE lights then press reconfigure button in Zha panel -> device picker

**Comment by comatose-tortoise** [comment]
I don't have any hue lights under the ZHA integration. 

Okay, so I did reconfigure one light. After this, I get a
`[0x3007:1:0x0006] Attribute report received: on_off=Bool.true`
every 15 minutes, only from this light, no other lights.

Seems promising!

I guess I'll press "reconfigure" for all lights (maybe all ZHA entities even). Follow up questions: What does this command do and why do I need to do this to get them to report? Will I have to do it regularly or is once enough?

**Comment by dmulcahey** [comment]
It issues the same commands issued when the devices join for the first time. You shouldn’t have to do it at all... we advise people to do it when things like this come up as a troubleshooting technique. It’s also helpful when we add support for new clusters so that people don’t have to remove and re-add devices to gain the new functionality.

**Comment by comatose-tortoise** [comment]
Ok, that's what I wanted to hear 👍

A bit weird that apparently none of my ZHA lights (all IKEA) are reporting attributes (including one I added this weekend). Is it possible to do something wrong when adding them through the ZHA interface? Seems like the only explanation..

BTW, what does the "ADD DEVICES" button do for the devices section in the ZHA interface? I don't mean the one on the top but the button that shows up after you select a device.

**Comment by Adminiuga** [comment]
"Add devices" button under the "device" allows joining of the new devices only through this particular device. Needed for Xiaomi, as it forces xiaomi to use the device you pressed "add devices" button to become xiaomi parent. Xiaomi are picky about their parent devices.

**Comment by Adminiuga** [comment]
Whether it is anything wrong adding them through the ZHA interface, you would need to check the logs. particularly for the "bind" results on On/off & Level clusters and results of "configure attribute reporting" on the same cluster.

**Comment by comatose-tortoise** [comment]
Ok! I reconfigured all my lights now, I'll report back later with results regarding lights showing the wrong state. Thanks a lot!

**Comment by comatose-tortoise** [comment]
Follow up: So reconfiguring the light definitely changed something, kind of towards the better. Now, the lights still "turns off" (no the actual light) in the, but are pretty quickly turned back on again. I'm guessing they are turned back on when their attribute report is sent to HA every 15th minute?

Here what the logbook said just now, for instance:

9:48 AM Tak hall turned on
9:45 AM Spot 2 turned on
9:45 AM Hörnlampa 2 turned on
9:45 AM Sideboardlampa turned on
9:45 AM Fönsterlampa bibliotek turned on
9:42 AM Fönsterlampa bibliotek turned off
9:42 AM Spot 2 turned off
9:42 AM Tak hall turned off
9:42 AM Hörnlampa 2 turned off
9:42 AM Sideboardlampa turned off

All of those are IKEA ZHA devices, and none of them actually turned off.

Any idea why they are doing this?

**Comment by Adminiuga** [comment]
No ideas but start with log checking. When the light turns off according to ha state machine, what was preceding that event?

**Comment by ErnstEeldert** [comment]
I have the same experience with ikea lights. They’re pretty much all over the place when it comes to state and responsiveness. Some seem to loose the network, although they will rejoin at some point, but that could take days. Toggling state in HA will cause the switch to flip back after 10 seconds or so.

Not very helpful, but just wanted to share :)

**Comment by Adminiuga** [comment]
What versions and hardware?
Same questions and troubleshooting steps as for comatose-turtoise.


> Toggling state in HA will cause the switch to flip back after 10 seconds or so

That is the usual behavior when you don't receive the ACK from the device to the command you send. Full debug logs would give you more hints

**Comment by comatose-tortoise** [comment]
Status report: Turned out it was only the 5 lights in the image in my previous post that was "turning off" in HA. They were doing this at the same time every hour:

![](https://i.imgur.com/OsNg4qM.png)

I just reconfigured these 5 lights (again) and after that, as you can see, they were acting normal and has since. 

There was nothing in the log that I could identify as weird before this happened. At least one of these lights sent an attribute report just two minutes before they were reported as off:
`2019-11-23 13:07:30 DEBUG (MainThread) [zigpy.zcl] [0x2f27:1:0x0006] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=False> manufacturer=None tsn=28 command_id=Command.Report_Attributes>
2019-11-23 13:07:30 DEBUG (MainThread) [zigpy.zcl] [0x2f27:1:0x0006] ZCL request 0x000a: [[<Attribute attrid=0 value=<TypeValue type=Bool, value=Bool.true>>]]
2019-11-23 13:07:30 DEBUG (MainThread) [zigpy.zcl] [0x2f27:1:0x0006] Attribute report received: on_off=Bool.true`

They were reported as OFF at 13:09, and the only thing I could find around that time that looked suspicious was this (same light as above):

`2019-11-23 13:09:00 DEBUG (MainThread) [homeassistant.components.zha.entity] light.fonsterlampa_kok: polling current state
2019-11-23 13:09:00 DEBUG (MainThread) [zigpy_deconz.zigbee.application] Sending Zigbee request with tsn 128 under 129 request id, data: b'0080000000'
2019-11-23 13:09:00 DEBUG (MainThread) [zigpy_deconz.api] Command Command.aps_data_request (20, 129, 0, <DeconzAddressEndpoint address_mode=2 address=0x2f27 endpoint=1>, 260, 6, 1, b'\x00\x80\x00\x00\x00', 2, 0)`

No idea what it means, though.

**Comment by Adminiuga** [comment]
It does exactly what the log says: `light.fonsterlampa_kok: polling current state`

**Comment by comatose-tortoise** [comment]
Well, it must have given the wrong answer then?

**Comment by Adminiuga** [comment]
🤷‍♂ only one way to find out. Post the debug logs.

**Comment by comatose-tortoise** [comment]
Ok, I will try to get a log of it when it happens the next time. Much more rare now after reconfiguration of those lights. 

One problem I have with the log is that there is a very large number of lines being generated constantly, so the logs gets huge. But I'll try to crop them around the time of the event.

Also, thank you for your patience, you are of great help!

**Comment by Adminiuga** [comment]
hrm not sure about docker, but I start my hass in venv with `--log-rotate-days 33` so it starts a new file every day. Gives me about 115MB log files with full ZHA debug enabled.

**Comment by comatose-tortoise** [comment]
Here is a log from tonight when the light light.sideboardlampa_vardagsrum (0x6a17) suddenly registered as being off at 22:36, and then back to on one hour later at 23:36. The light was on all this time.

https://www.dropbox.com/s/ujk9dctxjs6berk/homassistant.log?dl=1

As far as I understand, it isn't sending attribute reports, even not when being seen as on again? This light has been reconfigured quite recently. It's an IKEA light.

See anything informative in the log?

**Comment by comatose-tortoise** [comment]
This is still a very present issue for me, happens basically every day. You think it's possible it's not related to zigpy-deconz, but rather the ZHA integration?

**Comment by beeroo** [comment]
I've had the same issues with my ikea bulbs using ZHA and Conbee II. I was also forced to remove the ikea bulb from my daughter's room at wife's orders because it randomly turned on in the middle of the night and woke her up. Starting to think I'll have to splurge money on a HUE setup instead.. but it'd be nice if the IKEA stuff could be made reliable.

**Comment by comatose-tortoise** [comment]
What HA version are you on? This problem disappeared for me some time ago through an update. 

It's not perfect yet though, now instead they sometimes report as being on even though they are off.. :)

But the overall ZHA-deconz-IKEA situation is definately improved since I started this thread!

**Comment by Adminiuga** [comment]
> I've had the same issues with my ikea bulbs using ZHA and Conbee II [snipped] because it randomly turned on in the middle of the night and woke her up

How is it the same issue if OP is complaining on the state being incorrect and you are complaining on the light turning on randomly? Even without looking at any logs I can ensure you that ZHA integration does not turn on lights randomly. Not without being commanded to do so.

2nd, the message as is, is not aiding to the solution. Reproduce the issue and provide full debug log as per per https://www.home-assistant.io/integrations/zha/#reporting-issues in addition to data required by the issue template, otherwise it just wastes everyone's cycles.

**Comment by comatose-tortoise** [comment]
Just FYI: As far as I am concerned, this issue appears to be resolved as of 200705 (HA 112.2), in case you want to close it.
